backend/benefit/common/tests/test_utils.py

```diff
@@ -51,3 +59,72 @@ def test_days360_errors(date1, date2):
 )
 def test_date_range_overlap(start_1, end_1, start_2, end_2, expected_result):
     assert date_range_overlap(start_1, end_1, start_2, end_2) == expected_result
+
+
+def test_get_date_range_end_with_days360():
+
+    assert get_date_range_end_with_days360(date(2022, 1, 1), 0) is None
+    assert (
+        get_date_range_end_with_days360(date(2022, 1, 1), decimal.Decimal("0.01"))
+        is None
+    )
+    assert get_date_range_end_with_days360(
+        date(2022, 1, 1), decimal.Decimal("0.03")
+    ) == date(2022, 1, 1)
+    assert get_date_range_end_with_days360(
+        date(2022, 1, 1), decimal.Decimal("0.92")
+    ) == date(2022, 1, 28)
+    assert get_date_range_end_with_days360(
+        date(2022, 1, 1), decimal.Decimal("1")
+    ) == date(2022, 1, 31)
+    assert get_date_range_end_with_days360(
+        date(2022, 1, 1), decimal.Decimal("1.04")
+    ) == date(2022, 2, 1)
+    assert get_date_range_end_with_days360(
+        date(2022, 1, 20), decimal.Decimal("1.38")
+    ) == date(2022, 2, 28)
+    assert get_date_range_end_with_days360(
+        date(2022, 1, 30), decimal.Decimal("6.99")
+    ) == date(2022, 8, 30)
+    # special case
+    assert (
+        get_date_range_end_with_days360(date(2022, 2, 28), decimal.Decimal("0.04"))
+        is None
+    )
```

Comment:
This is more compact with parametrizing:
```python
@pytest.mark.parametrize(
    "date1,date2,expected_result",
    [
        (date(2022, 1, 1), 0, None),
        (date(2022, 1, 1), decimal.Decimal("0.01"), None),
        (date(2022, 1, 1), decimal.Decimal("0.03"), date(2022, 1, 1)),
        (date(2022, 1, 1), decimal.Decimal("0.92"), date(2022, 1, 28)),
        (date(2022, 1, 1), decimal.Decimal("1"), date(2022, 1, 31)),
        (date(2022, 1, 1), decimal.Decimal("1.04"), date(2022, 2, 1)),
        (date(2022, 1, 20), decimal.Decimal("1.38"), date(2022, 2, 28)),
        (date(2022, 1, 30), decimal.Decimal("6.99"), date(2022, 8, 30)),
        (date(2022, 2, 28), decimal.Decimal("0.04"), None),
    ],
)
def test_get_date_range_end_with_days360(date1, date2, expected_result):
    assert get_date_range_end_with_days360(date1, date2) == expected_result
```

This would require moving the `if os.getenv("DAYS_360_FULL_TEST") == "1":` part to another test function.

Reply:
good idea